app/javascript/packs/admin/editUser.jsx

```diff
@@ -2,6 +2,20 @@ import { initializeDropdown } from '@utilities/dropdownUtils';
 
 let preact;
 let AdminModal;
+const modalContents = new Map();
+
+function getModalContents(modalContentSelector) {
```

Comment:
We cached the contents of the rendered partial for a specific modal and then remove the node from the DOM so we do not get duplicate ID errors in Axe.

Reply:
I don't see any problem with this approach from an a11y POV, but does this work if, for example,

- I open modal 1
- I close it and open modal 2
- I try to open modal 1 again?

Or will this function go off looking for the hidden modal contents div and not find it because we deleted it during step 1?

Reply:
No, it caches it in client-side code when it first removes it from the DOM. From there no matter how many times you open that specific modal, the contents will be pulled from the cache (the Map).